oxd-server/src/main/java/org/gluu/oxd/server/service/DiscoveryService.java

```diff
@@ -60,7 +61,43 @@ public OpenIdConfigurationResponse getConnectDiscoveryResponseByOxdId(String oxd
     }
 
     public OpenIdConfigurationResponse getConnectDiscoveryResponse(Rp rp) {
-        return getConnectDiscoveryResponse(rp.getOpHost(), rp.getOpDiscoveryPath());
+        return getConnectDiscoveryResponse(rp.getOpConfigurationEndpoint(), rp.getOpHost(), rp.getOpDiscoveryPath());
+    }
+
+    public OpenIdConfigurationResponse getConnectDiscoveryResponse(String opConfigurationEndpoint, String opHost, String opDiscoveryPath) {
```

Comment:
This method is redundant. `opConfigurationEndpoint` = `opHost` + `opDiscoveryPath`.

We don't need copy paste. Lets stick to single method  (refactor existing method and re-use)

```
public OpenIdConfigurationResponse getConnectDiscoveryResponse(String opConfigurationEndpoint)
```

Reply:
Done